### ISSUE TITLE
auto formatting even when both formatOnSave and formatOnType are set to false

### ISSUE THREAD

**Issue by tzuriel-spearuav** [microsoft/vscode-cpptools#6107]
Type: General

Working on a QT application, I have a .cc file that I'm editing and noticed upon save it generates a lot of editing to the file - such as indentation. This is unwanted behavior as I'm trying to do the most minimal changes on my forked repository while origin keeps getting updated.

My vscode preferences are set as follows :
	"editor.formatOnSave": false
	"editor.formatOnType": false

I'm running an updated vscode (1.48.2), with v0.29.0 of the cpp extension on a Win10 laptop.
The above behavior is not happening after disabling cpp extension

**Comment by sean-mcmanus** [comment]
Do you have a remote, workspace, or workspace folder setting that uses a different formatOnSave value?

Can you set C_Cpp.loggingLevel to "Debug" and see what formatting messages are being reported?

We're not able to repro this so far.

**Comment by tzuriel-spearuav** [comment]
I don't have a local settings.json (but made sure by creating one at the root, see attached)
All other extensions are disabled.

I created a very simple cpp file (so unrelated to file extension) with the following content:
`const char* Class::units =  <<multiple spaces/tabs>> "km";`
Saving it will format it as follows:
`const char *Class::units = "km";`
(formatting moved the asterisk pointer and removed white spaces)

After setting the debug flag you indicated, and saving the cpp file, I get the following:
```
idle loop: reparsing the active document
Checking for syntax errors: file:///c%3A/code/settings.cpp
Queueing IntelliSense update for files in translation unit of: C:\CODE\SETTINGS.CPP
Error squiggle count: 1
Update IntelliSense time (sec): 0.256
Formatting input:
Formatting document: file:///c%3A/code/settings.cpp
Formatting raw output:
Formatting diffed output:
  tag parsing file: C:\CODE\SETTINGS.CPP
idle loop: reparsing the active document
Checking for syntax errors: file:///c%3A/code/settings.cpp
Queueing IntelliSense update for files in translation unit of: C:\CODE\SETTINGS.CPP
sending 3 changes to server
Error squiggle count: 1
Update IntelliSense time (sec): 0.267
Checking for syntax errors: file:///c%3A/code/settings.cpp
Queueing IntelliSense update for files in translation unit of: C:\CODE\SETTINGS.CPP
Error squiggle count: 1
Update IntelliSense time (sec): 0.26
```

BTW, peeking into the output pane, under 'Log (extension Host)', I see quite a few of the following:
```
[2020-09-11 00:40:32.472] [exthost] [warning] ms-vscode.cpptools - Code actions of kind 'quickfix 'requested but returned code action does not have a 'kind'. Code action will be dropped. Please set 'CodeAction.kind'.
[2020-09-11 00:40:35.868] [exthost] [error] [ms-vscode.cpptools] provider FAILED
[2020-09-11 00:40:35.868] [exthost] [error] undefined
```
![settings_json](https://user-images.githubusercontent.com/70841703/92813884-d0c90600-f3ca-11ea-8550-ec9972064edc.png)

**Comment by sean-mcmanus** [comment]
Your logs appear incomplete.

Can you include the messages for "textDocument/formatting", "textDocument/didChange", and "textDocument/didSave"?

**Comment by tzuriel-spearuav** [comment]
He're the full log - 
```
cpptools/didChangeCppProperties
$/setTraceNotification
cpptools/getCodeActions: file:///c%3A/code/settings.cpp (id: 94)
cpptools/activeDocumentChange: file:///c%3A/code/settings.cpp
cpptools/textEditorSelectionChange
cpptools/getDocumentSymbols: file:///c%3A/code/settings.cpp (id: 95)
cpptools/textEditorSelectionChange
cpptools/getDocumentSymbols
cpptools/getFoldingRanges: file:///c%3A/code/settings.cpp (id: 96)
cpptools/getCodeActions: file:///c%3A/code/settings.cpp (id: 97)
idle loop: reparsing the active document
Checking for syntax errors: file:///c%3A/code/settings.cpp
Queueing IntelliSense update for files in translation unit of: C:\CODE\SETTINGS.CPP
cpptools/finishUpdateSquiggles
Error squiggle count: 1
Update IntelliSense time (sec): 0.256
cpptools/getSemanticTokens: file:///c%3A/code/settings.cpp (id: 98)
cpptools/getSemanticTokens: file:///c%3A/code/src/Settings/AppSettings.cc (id: 99)
cpptools/getCodeActions: file:///c%3A/code/settings.cpp (id: 100)
textDocument/formatting: file:///c%3A/code/settings.cpp (id: 101)
Formatting input:
const char* Class::units =              "km";
Formatting document: file:///c%3A/code/settings.cpp
Formatting raw output:
const char *Class::units = "km";
Formatting diffed output:
 *Class::units = "km";
textDocument/didChange: file:///c%3A/code/settings.cpp
cpptools/textEditorSelectionChange
textDocument/didSave: file:///c%3A/code/settings.cpp
  tag parsing file: C:\CODE\SETTINGS.CPP
cpptools/fileChanged: file:///c%3A/code/settings.cpp
cpptools/getFoldingRanges: file:///c%3A/code/settings.cpp (id: 103)
cpptools/getCodeActions: file:///c%3A/code/settings.cpp (id: 104)
cpptools/getSemanticTokens: file:///c%3A/code/settings.cpp (id: 105)
idle loop: reparsing the active document
Checking for syntax errors: file:///c%3A/code/settings.cpp
Queueing IntelliSense update for files in translation unit of: C:\CODE\SETTINGS.CPP
cpptools/getDocumentSymbols: file:///c%3A/code/settings.cpp (id: 106)
cpptools/getDocumentSymbols
sending 3 changes to server
cpptools/finishUpdateSquiggles
Error squiggle count: 1
Update IntelliSense time (sec): 0.26
cpptools/getCodeActions: file:///c%3A/code/settings.cpp (id: 107)
Checking for syntax errors: file:///c%3A/code/settings.cpp
Queueing IntelliSense update for files in translation unit of: C:\CODE\SETTINGS.CPP
cpptools/finishUpdateSquiggles
Error squiggle count: 1
Update IntelliSense time (sec): 0.256
cpptools/getSemanticTokens: file:///c%3A/code/settings.cpp (id: 108)
cpptools/getSemanticTokens: file:///c%3A/code/src/Settings/AppSettings.cc (id: 109)
cpptools/getCodeActions: file:///c%3A/code/settings.cpp (id: 110)
```

**Comment by sean-mcmanus** [comment]
Those logs look like VS is sending us a textDocument/formatting LSP message as a result of textDocument/didSave.

If you create a test.js file with `var i =            0;` and do a save, does it format in that case too? If so, you should file a bug with VS Code at https://github.com/microsoft/vscode/issues .

You may also want to try https://github.com/microsoft/vscode-cpptools/releases/tag/0.30.0-insiders5 , because we changed the implementation to use different formatting messages.

**Comment by sean-mcmanus** [comment]
@tzuriel-spearuav Also, if you're able to locate which version of VS Code or the C/C++ extension started reproducing the bug, that might help. The willSaveWaitUntil code hasn't changed for over a year and the formatting code hasn't changed much either, up until 1.0.0 that will be released today. Also, if you're able to repro the issue on other machines or other workspaces that might help identify what is different.

**Comment by sean-mcmanus** [comment]
@tzuriel-spearuav Another thing you could try is to clone https://github.com/microsoft/vscode-cpptools.git, open the Extension subfolder and do Launch Extension (development), and then try to debug any formatting or willSaveWaitUntil code, such as vscode-cpptools\Extension\node_modules\vscode-languageclient\lib\client.js line 539 that does `let vEdits = this._client.protocol2CodeConverter.asTextEdits(edits);` and make sure no edits are being received back (we send null), or putting breakpoints in the formatting related handlers in that file and looking up the call stack to see what is causing that formatting to get incorrectly invoked.

**Comment by tzuriel-spearuav** [comment]
@sean-mcmanus cannot reproduce this issue after re-installing vscode, neither on other systems. So I'm currently unable to reproduce any longer

**Comment by sean-mcmanus** [comment]
Okay, that sounds good. Might have been some issue with your install.

We just published 1.0.0 of our extension so that might have something to do with it too.
